source/blog/2015-03-22-rom-0-6-0-released.html.markdown

```diff
@@ -0,0 +1,186 @@
+* [rom-rails 0.3.0](https://rubygems.org/gems/rom-rails) - [changelog](https://github.com/rom-rb/rom-rails/blob/master/CHANGELOG.md#v040-2015-03-22)
+
+Following new adapters have been released too:
+
+* [rom-yesql](https://rubygems.org/gems/rom-yesql)
+* [rom-csv](https://rubygems.org/gems/rom-csv)
+* [rom-yaml](https://rubygems.org/gems/rom-yaml)
+
+## Class-based setup
+
+The biggest change in this release is the ability to define your own relation, mapper and command subclasses that ROM will instantiate for you and expose through its registry. The DSL-based setup is still available but using classes turned out to be a better approach in bigger contexts like web-applications.
+
+Here's a simple Relation subclass definition:
+
+``` ruby
+class Users < ROM::Relation[:memory]
+  def by_name(name)
+    where(name: name)
+  end
+end
+```
+
+## Lazy-relations with data-pipelining
+
+Previous version of ROM had the concept of Reader which turned out to be confusing. After lots of discussions it was replaced by a new interface for accessing relations. This change makes mapping relations explicit as you need to provide the name of mapper(s) that you want to use when loading a relation.
+
+Lazy relations can be used with any object that implements mapper interface but the simplest way to levarage this feature is to register ROM mappers for quick access:
+
+``` ruby
+ROM.setup(:memory)
+
+class Users < ROM::Relation[:memory]
+  def by_name(name)
+    restrict(name: name)
+  end
+end
+
+class UserMapper < ROM::Mapper
+  relation :users
+  register_as :entity
+
+  model User
+
+  attribute :name
+end
+
+rom = ROM.finalize.env
+
+user_entites = rom.relation(:users).as(:entity)
+user_entities.by_name('Jane')
+```
+
+Thanks to data-pipelining feature you can send a relation through multiple mapping objects. This turned out to be very useful for decorating data:
+
+``` ruby
+class UserPresenterMapper < ROM::Mapper
+  relation :users
+  register_as :presenter
+
+  model UserPresenter
+end
+
+user_presenter = rom.relation(:users).as(:entity, :presenter).by_name('Jane').first
+```
+
+On top of that there's also an experimental partial-application feature where you can grab a reference to a relation with partially applied arguments:
+
+``` ruby
+users_by_name = rom.relation(:users).as(:entity).by_name
+
+users_by_name['Jane']
+```
+
+## Improved and simplified command interface
+
+Commands are now composable and you can access them directly and call them. This makes it more obvious what's going on. We are still working on a generalized error/success handling that will be based on a 3rd party dependency since this is really out of ROM's scope.
+
+You can define your commands using class interfaces too:
+
+``` ruby
+ROM.setup(:memory)
+
+class CreateUser < ROM::Commands::Create[:memory]
+  relation :users
+  register_as :create
+  result :one
+end
+
+rom = ROM.finalize.env
+
+create_user = rom.commands.users[:create]
```

Comment:
`rom.commands.users[:create]`

Should this be `rom.commands.users(:create)`? Or does `[]` work as well?